be/src/storage/tablet_manager.cpp

```diff
@@ -650,6 +650,11 @@ TabletSharedPtr TabletManager::find_best_tablet_to_do_update_compaction(DataDir*
                 continue;
             }
 
+            if (!tablet_ptr->updates()->get_update_compaction_lock().try_lock()) {
+                continue;
+            }
+            tablet_ptr->updates()->get_update_compaction_lock().unlock();
```

Comment:
why unlock here?

Reply:
In the `find_best_tablet_to_do_update_compaction` function here, the tablet selected here is just a candidate, it is not necessarily the tablet that will eventually perform update compaction, thus i will unlock the candidate tablet here.

Reply:
It seems not very necessary. Base/Cumulative compaction are not got a lock in this place. They all get a lock in compact().

Reply:
It seems that Base/Cumulative compaction also got a lock in this place.